crypto/util/pem.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package util
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+)
+
+// PemToPublicKey converts a PEM encoded public key to a crypto.PublicKey
+func PemToPublicKey(pub []byte) (crypto.PublicKey, error) {
+	block, _ := pem.Decode(pub)
+	if block == nil || block.Type != "PUBLIC KEY" {
```

Comment:
I think the header can also be specified with the type e.g. `RSA PUBLIC KEY` or `EC PUBLIC KEY` so it might not just be `public key`.

Reply:
added other options as well.

Reply:
although EC public key is always marshalled as ----BEGIN PUBLIC KEY------